src/core/qgsvectorlayer.h

```diff
@@ -997,6 +997,13 @@ class CORE_EXPORT QgsVectorLayer : public QgsMapLayer, public QgsExpressionConte
      */
     long featureCount( const QString &legendKey ) const;
 
+    /**
+     * Ids of features rendered with specified legend key. Features must be first
+     * calculated by countSymbolFeatures()
+     * \returns Ids of features rendered by symbol or -1 if failed or Ids are not available
```

Comment:
Nearly there!

Last change - can you add

    \since QGIS 3.10

Here?

Reply:
Should be done, took far longer than I wanted.